app/Http/Controllers/CompetitionsController.php

```diff
@@ -94,15 +95,38 @@ public function destroy(Competition $competition)
     }
 
     /**
-     * Download the CSV export of all users.
+     * Download a CSV export of all users based on different criteria.
      *
      * @param  \Gladiator\Models\Competition  $competition
-     * @return \League\Csv\ $csv
+     * @param  \Illuminate\Http\Request  $request
+     * @return void
      */
-    public function export(Competition $competition)
+    public function export(Competition $competition, Request $request)
     {
-        $csv = $this->manager->exportCSV($competition);
-        $csv->output('competition' . $competition->id . '.csv');
+        $fileName = 'contest_' . $competition->contest_id . '-' . 'competition_' . $competition->id;
+
+        $withReportback = convert_string_to_boolean($request->input('withReportback'));
+
+        $users = $this->manager->getModelUsers($competition, $withReportback);
+
+        if (! $withReportback) {
+            // @TODO: can probably flash this to the session too!
+            $fileName = $fileName . '-users';
+        } else {
+            $fileName = $fileName . '-leaderboard';
+
+            if (session()->has($fileName)) {
```

Comment:
When the leaderboard page loads that is request #1, the leaderboard view shows and we see the leaderboard, end of execution. If we then export the leaderboard as a CSV, that is request #2, it would need to go and do all the DB & API queries all over again to grab that info and then build the CSV to download (what we were doing before). Instead, I flash all that data to the session and it persists in a very very temporary cache and is available just for request #2. Super fast export of leaderboard!!